### ISSUE TITLE
Broken TypeScript file generated from trivial app

### ISSUE THREAD

**Issue by muelli** [dillonkearns/elm-typescript-interop#4]
A trivial app like

```
module MyApp exposing (..)

main = text "Hello"
```

does not fail to compile, but causes a broken typescript file to be generated:

```
// WARNING: Do not manually modify this file. It was generated using:
// https://github.com/dillonkearns/elm-typescript-interop
// Type definitions for Elm ports
export as namespace Elm


export interface App {
  ports: {

  }
}
    

export namespace NOT FOUND {
  export function fullscreen(): App
  export function embed(node: HTMLElement | null): App
}
```

I expected to either see the interop compiler fail or get a functional TypeScript file.

**Comment by dillonkearns** [comment]
@muelli thanks for filing this bug report!

To fill you in on my investigation, there are a couple of issues:
1) There needs to be a type annotation, `elm-typescript-interop` should give an error message for the missing type annotation.
2) Even if there is an annotation, I'm assuming that there will be a function called `main` of type `Program`. But having Html is valid and should be supported.

I'll keep you posted on my progress here.

**Comment by dillonkearns** [comment]
I pushed a fix for 1 on master (not yet published). With your above example, the output is now
```
Error parsing input file src/Main.elm

No main function with type annotation found.
```

**Comment by dillonkearns** [comment]
Alright, I've added support for main functions with type Html, and I've made the messaging more clear if there is a failure due to a missing type annotation. I've verified that the example you shared works when the type annotation is added (and gives the proper message otherwise).

Thanks again for filing this issue!
